org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/SnippetCompletionProposal.java

```diff
@@ -35,34 +51,143 @@
 	private static final String INTERFACE_SNIPPET_LABEL = "interface";
 	private static final Set<String> UNSUPPORTED_RESOURCES = Sets.newHashSet("module-info.java", "package-info.java");
 
-	public static List<CompletionItem> getSnippets(ICompilationUnit cu) {
+	public static List<CompletionItem> getSnippets(ICompilationUnit cu, CompletionContext completionContext, IProgressMonitor monitor) {
 		if (cu == null) {
 			throw new IllegalArgumentException("Compilation unit must not be null"); //$NON-NLS-1$
 		}
+		if (monitor == null) {
+			monitor = new NullProgressMonitor();
+		}
 		//This check might need to be pushed back to the different get*Snippet() methods, depending on future features
 		if (UNSUPPORTED_RESOURCES.contains(cu.getResource().getName())) {
 			return Collections.emptyList();
 		}
+		if (!accept(cu, completionContext, true)) {
```

Comment:
L65-L70 need to be moved inside getClassSnippet.

if accept returns false, then you're not even trying interfaces next